protocol/x/clob/memclob/memclob.go

```diff
@@ -1352,11 +1352,17 @@ func (m *MemClobPriceTimePriority) validateNewOrder(
 		return types.ErrInvalidReplacement
 	}
 
-	// If the order is a reduce-only order, we should ensure it does not increase the subaccount's
-	// current position size. Note that we intentionally do not validate that the reduce-only order
+	// If the order is a reduce-only order, we should ensure the full size of the order does not increase the
+	// subaccount's current position size. Note that we intentionally do not validate that the reduce-only order
 	// does not change the subaccount's position _side_, and that will be validated if the order is matched.
 	// TODO(DEC-1228): use `MustValidateReduceOnlyOrder` and move this to `PerformStatefulOrderValidation`.
 	if order.IsReduceOnly() {
+		// If there exists an non-reduce only order with the same order id
+		// cancel the incoming reduce only order by returning an error.
+		if restingOrderExists && !existingRestingOrder.IsReduceOnly() {
+			return types.ErrReduceOnlyOrderReplacement
+		}
```

Comment:
why is this logic necessary?

Reply:
Started discussion in #p-reduce-only-ioc-fok